src/components/AIAnalyticsSummary/data/hooks.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import { logError } from '@edx/frontend-platform/logging';
+import LmsApiService from '../../../data/services/LmsApiService';
+
+const useAIAnalyticsSummary = (enterpriseId, insights) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [analyticsSummaryData, setAnalyticsSummaryData] = useState(null);
```

Comment:
I think it would be even better to use something like
```
const [error, setError] = useState(null);
```

And set the error value of there is an error while making the API call. And show the error to the user.